tools/percy/snapshots.js

```diff
@@ -25,15 +31,23 @@ const pagesToTest = [
 
 // A script to navigate our app and take snapshots with Percy.
 PercyScript.run(
-  async (browser, percySnapshot) => {
-    for (page of pagesToTest) {
-      await browser.goto(`http://localhost:8080/${page.url}`);
-      await browser.evaluate(scrollToBottom);
-      await browser.waitFor(2000);
-      await percySnapshot(`${page.title}`);
+  async (/** @type {!puppeteer.Page} */ page, percySnapshot) => {
+    for (const {url, title} of pagesToTest) {
+      await page.goto(`http://localhost:8080/${url}`);
+      await page.evaluate(scrollToBottom);
+      await page.waitFor(2000);
+      await percySnapshot(title);
     }
+    // Navigate back to home page by clicking on a link. We expect most assets
+    // to already be lazyloaded by this point so don't scroll.
+    // This is a canary for problems with the SPA routing code.
+    const navigationPromise = page.waitForNavigation();
+    await page.click(".web-header__logo-link");
+    await navigationPromise;
+    await page.waitFor(200);
+    await percySnapshot("Home page, navigation");
```

Comment:
I actually don't know if this page would have shown the failure we were seeing before. It seemed like it was only happening to content images inside of blog posts. I'm curious if it's worth clicking through a few pages and snapshotting them since the snapshots are fairly cheap. Could go Home > Blog > Post

Reply:
Come to think of it... we could just click through the site and snapshot the pages, instead of going directly to the urls. That way we'd test SPA and snapshot all the pages at the same time.

Reply:
So I think you might be right, in that clicking this will still actually work if the JS isn't running. I need to think about this a bit more.

Reply:
That's true but I think clicking through the pages would have caught the failures we had the other day where images were being request wrong. I think it's worth adding these tests and continuing to improve on them. But I would like to click through the blog and at least one post to snapshot those pages as well because they're so critical.